presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -227,6 +236,17 @@ public TaskStatus process()
                     return TaskStatus.finished();
                 }
                 ListenableFuture<?> future;
+                /*
+                 * Block until one of below conditions is met:
+                 * Arrival of first non "all" filter
```

Comment:
I don't think it's needed. We short-circut on all predicate.

Reply:
Didn't follow this completely, my intent is to unblock the probe side scheduling as soon as we have a useful (not `all`) filter available from DynamicFilter without waiting for full completion. The future from `isBlocked` can complete multiple times with an `all` before giving a not `all` filter, so I used the condition `!dynamicFilter.isComplete() && dynamicFilter.getCurrentPredicate().isAll()` before blocking on the future from `dynamicFilter.isBlocked()`